kernel/src/transaction.rs

```diff
@@ -127,6 +156,16 @@ impl Transaction {
         self
     }
 
+    /// Include a SetTransaction (app_id and version) action for this transaction.
+    /// Note that each app_id can only appear once per transaction. That is, multiple app_ids with
+    /// different versions are disallowed in a single transaction. If a duplicate app_id is
+    /// included, the `commit` will fail (that is, we don't eagerly check app_id validity here).
+    pub fn with_transaction_id(mut self, app_id: String, version: i64) -> Self {
```

Comment:
```suggestion
    pub fn with_transaction_id(
        mut self, 
        app_id: impl Into<String>, 
        version: i64,
        last_updated: impl Into<Option<i64>>,
    ) -> Self {
```
Easier to use that way?
```rust
txn
    .with_transaction_id("app_id1", 1, None)
    .with_transaction_id("app_id2", 1, Some(1744732734))
```

Reply:
yea was curious how we wanted to handle the 'last_updated' - I supposed we could provide the ultimate flexibility to the engine by going this way? (another alternative that may be too prescriptive would be us trying to fill that ourselves? but I worry that means we are making too many decisions about what timestamp to store there?)

Reply:
Yeah, the interpretation is 100% up to the user (not even the engine)

Reply:
Actually, maybe it's not... the [spec](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#transaction-identifiers) says it's:
> The time when this transaction action is created, in milliseconds since the Unix epoch

Based on that, kernel _should_ set it, not trusting user.

Reply:
yup sounds good - i've updated to this API here :)

Reply:
oh - race condition on these comments I think haha. hmm then I suppose old API and we just grab a `SystemTime::now().duration_since(UNIX_EPOCH)` to put into last_updated?

Reply:
looks like java kernel does this at the builder level to keep all the `txn` actions in the same transaction consistently timestamped? https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java#L120

Reply:
Yeah, for now let's be consistent with the java kernel (and with the spec)

Reply:
i'll make these modifications and merge! thanks!

Reply:
Honestly, we should probably use the in-commit timestamp for this (and also `remove.deletionTimestamp`, once we eventually support removes)